controllers/mattermost/mattermost/mattermost.go

```diff
@@ -20,55 +21,83 @@ import (
 	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+type reconcileStatus struct {
+	Status bool
+	Error  error
+}
+
 func (r *MattermostReconciler) checkMattermost(
 	mattermost *mmv1beta.Mattermost,
 	dbInfo mattermostApp.DatabaseConfig,
 	fileStoreInfo *mattermostApp.FileStoreInfo,
 	status *mmv1beta.MattermostStatus,
-	reqLogger logr.Logger) error {
+	reqLogger logr.Logger) reconcileStatus {
 	reqLogger = reqLogger.WithValues("Reconcile", "mattermost")
 
+	recStatus := reconcileStatus{
+		Status: false,
+		Error:  nil,
+	}
+
 	err := r.checkLicence(mattermost)
-	if err != nil {
-		return errors.Wrap(err, "failed to check mattermost license secret.")
+	if err.Error != nil {
+		recStatus.Error = errors.Wrap(err.Error, "failed to check mattermost license secret.")
+		return recStatus
 	}
 
 	err = r.checkMattermostService(mattermost, status, reqLogger)
-	if err != nil {
-		return err
+	if err.Error != nil {
+		recStatus.Error = err.Error
+		return recStatus
 	}
 
 	err = r.checkMattermostRBAC(mattermost, reqLogger)
-	if err != nil {
-		return err
+	if err.Error != nil {
+		recStatus.Error = err.Error
+		return recStatus
 	}
 
 	if !mattermost.Spec.UseServiceLoadBalancer {
 		err = r.checkMattermostIngress(mattermost, reqLogger)
-		if err != nil {
-			return err
+		if err.Error != nil {
+			recStatus.Error = err.Error
+			return recStatus
 		}
 	}
 
 	err = r.checkMattermostDeployment(mattermost, dbInfo, fileStoreInfo, status, reqLogger)
-	if err != nil {
-		return err
+	if err.Error != nil {
+		recStatus.Error = err.Error
+		return recStatus
 	}
 
-	return nil
+	return recStatus
 }
 
-func (r *MattermostReconciler) checkLicence(mattermost *mmv1beta.Mattermost) error {
+func (r *MattermostReconciler) checkLicence(mattermost *mmv1beta.Mattermost) reconcileStatus {
```

Comment:
I think it is enough if we return `reconcileStatus` only from `checkMattermostDeployment` and `checkMattermost`. Lets keep the remaining functions simple as long as we do not need it.

Reply:
Okay Thanks. We also need it to return from `updateMattermostDeployment`